src/nvim/lua/executor.c

```diff
@@ -2177,3 +2177,26 @@ char *nlua_funcref_str(LuaRef ref)
   kv_printf(str, "<Lua %d>", ref);
   return str.items;
 }
+
+char *nlua_read_secure(const char *path)
+{
+  lua_State *const lstate = global_lstate;
+  lua_getglobal(lstate, "vim");
+  lua_getfield(lstate, -1, "secure");
+  lua_getfield(lstate, -1, "read");
+  lua_pushstring(lstate, path);
+  lua_call(lstate, 1, 1);
+
+  size_t len = 0;
+  const char *contents = lua_tolstring(lstate, -1, &len);
+  char *buf = NULL;
+  if (contents != NULL) {
+    buf = xcalloc(len, sizeof(char));
```

Comment:
I'm not sure about this, but I think `len` might not include the null byte.

Reply:
It does not. Fixed in d1a847ca9a7323db1b74a2fd77ee8b351be1696d